### PR TITLE
Removed FileDescriptor and modified PipeUnix to match

### LINKED ISSUE #966: Replace FileDescriptor with File in PipeUnix
`FileDescriptor` has been deprecated and the only remaining use is in `PipeUnix`, after that it can be removed.

### PR DESCRIPTION
**Pull request by marcusnaslund**
Fixes #966

### CODE REVIEW

**Comment by marcusnaslund** [comment]
@sebastianbaginski - do you know what `close` does? It doesn't even take an argument. Also, we don't call it where we use this in the other project.

File: source/io/native/PipeUnix.ooc [hunk 1]
```diff
@@ -1,77 +1,72 @@
...
-		fd close()
+		0
+		//fd := _getFD(end)
+		//fd == 0 ? 0 : fd close()
 	}
 	close: func ~both {
-		readFD close()
-		writeFD close()
+		//readFD close()
+		//writeFD close()
```
Comment (marcusnaslund) [inline source/io/native/PipeUnix.ooc]: @sebastianbaginski - do you know what `close` does? It doesn't even take an argument. Also, we don't call it where we use this in the other project.

Comment (sebastianbaginski) [inline source/io/native/PipeUnix.ooc]: I think it calls `close(fd)` on a FileDescriptor object. Basically it just closes the file. See `man close`.

Comment (sebastianbaginski) [inline source/io/native/PipeUnix.ooc]: So why did you remove it from here ? Also I think we should call it in the other project. Always clean up your stuff, you know. But the pipe will be closed by the OS when the process exits anyway, so not really a big problem.

Comment (marcusnaslund) [inline source/io/native/PipeUnix.ooc]: > So why did you remove it from here ?

Oh, I'm not finished, hence no review flag. :) Just wanted your input. My point is that there is no `close(fd)`, just `close()`, so what does it really do? Or does rock throw in the argument automagically?

Comment (sebastianbaginski) [inline source/io/native/PipeUnix.ooc]: I don't know, it should be more like `close(this)` in `FileDescriptor`

**Comment by marcusnaslund** [comment]
Updated and ready for review.
